newIDE/app/src/ResourcesList/FileToCloudProjectResourceUploader.js

```diff
@@ -0,0 +1,206 @@
+        });
+      } finally {
+        setIsUploading(false);
+      }
+    },
+    [
+      selectedFiles,
+      authenticatedUser,
+      onChooseResources,
+      createNewResource,
+      cloudProjectId,
+    ]
+  );
+
+  const invalidFiles = selectedFiles
+    .map(file => {
+      if (file.size > PROJECT_RESOURCE_MAX_SIZE_IN_BYTES) {
+        return {
+          filename: file.name,
+          error: 'too-large',
+        };
+      }
+      return null;
+    })
+    .filter(Boolean);
+
+  const canUploadWithThisStorageProvider =
+    storageProvider.internalName === 'Cloud';
+  const isConnected = !!authenticatedUser.profile;
+  const canChooseFiles =
+    !isUploading && isConnected && canUploadWithThisStorageProvider;
+  const canUploadFiles =
+    !isUploading &&
+    canChooseFiles &&
+    hasSelectedFiles &&
+    invalidFiles.length === 0;
+
+  return (
+    <ColumnStackLayout noMargin>
+      {!isConnected ? (
+        <AlertMessage kind="warning">
+          <Trans>
+            Your need to first create your account, or login, to upload your own
+            resources.
+          </Trans>
+        </AlertMessage>
+      ) : !canUploadWithThisStorageProvider ? (
+        <AlertMessage kind="warning">
+          <Trans>
+            Your need to first save your game on GDevelop Cloud to upload your
+            own resources.
+          </Trans>
+        </AlertMessage>
+      ) : null}
+      <Paper variant="outlined">
+        <Line expand>
+          <Column expand>
+            <input
+              accept={resourceKindToInputAcceptedFiles[options.resourceKind]}
+              style={styles.fileInput}
+              multiple={options.multiSelection}
+              type="file"
+              ref={inputRef}
+              disabled={!canChooseFiles}
+              onChange={event => {
+                const files = [];
+                for (let i = 0; i < event.currentTarget.files.length; i++) {
+                  files.push(event.currentTarget.files[i]);
+                }
+                setSelectedFiles(files);
+              }}
+            />
+          </Column>
+        </Line>
+      </Paper>
+      {invalidFiles.map(erroredFile => {
+        if (erroredFile.error === 'too-large')
+          return (
+            <AlertMessage kind="error">
+              <Trans>
+                The file {erroredFile.filename} is too large. Use files that are
+                smaller for your game.
+              </Trans>
+            </AlertMessage>
+          );
+
+        return (
+          <AlertMessage kind="error">
+            <Trans>The file {erroredFile.filename} is invalid.</Trans>
+          </AlertMessage>
+        );
+      })}
+      <LineStackLayout alignItems="center" justifyContent="flex-end" expand>
+        {isUploading ? (
+          <LinearProgress expand value={uploadProgress} variant="determinate" />
+        ) : null}
+        <RaisedButton
+          onClick={onUpload}
+          disabled={!canUploadFiles}
```

Comment:
Should this be disabled when uploading as well?

Reply:
It is :)